### DISCUSSION TITLE
Handle audio ml:Fields using sc:AudioObject

### DISCUSSION THREAD

**Discussion by marcenacp** [mlcommons/croissant#242] [Ideas]
**Idea**: We want to handle audio ml:Fields using https://schema.org/AudioObject.

## Issue 1

Does sc:AudioObject contain all the needed properties? In particular, is `bitrate` being an sc:Text an issue?

## Issue 2

The other challenge is to have `ml:Field` inherit all the properties of `sc:AudioObject`.

**Solution 1**: Add a field `audioProperties` with all the audio properties.

```json
{
  "@type": "ml:Field",
  "name": "audio",
  ...
  "audioProperties": {
    "@type": "sc:AudioObject",
    "bitrate": "24kbps",
    "encodingFormat": "audio/mpeg"
  }
}
```

**Solution 2**: Use [additionalProperty](https://schema.org/additionalProperty).

```json
{
  "@type": "ml:Field",
  "name": "audio",
  ...
  "additionalProperty": [
    {
      "@type": "sc:PropertyValue",
      "name": "bitrate",
      "value": "24kbps"
    },
    {
      "@type": "sc:PropertyValue",
      "name": "encodingFormat",
      "value": "audio/mpeg"
    }
  ]
}
```

This discussion is related to the issue https://github.com/mlcommons/croissant/issues/240. Cc: @benjelloun @monke6942021 @fineguy.

**Comment by fineguy** [comment]
I think having `bitrate` as text field is fine, because it usually includes measure units, e.g. "320Kbps" (common for MP3 files) or "10Mpbs" (common for Full HD videos).

But I'm not sure if `sc:AudioObject` describes audio data well enough for data processing tasks. For example I don't see anything related to sampling rate or number of channels, which are commonly necessary for correct audio file reading, e.g. in [`librosa.load`](https://librosa.org/doc/latest/generated/librosa.load.html#librosa.load) or [`pydub.AudioSegment.from_file`](https://github.com/jiaaro/pydub/blob/master/API.markdown#audiosegmentfrom_file).

**Reply by marcenacp** [reply]
@fineguy Thanks for the answer.

@benjelloun sc:AudioObject doesn't have all the required properties for data processing. What is the best practice? Should we create ml:AudioObject or extend sc:AudioObject with the additional properties (`numChan`, `dtype`)?

**Comment by benjelloun** [comment]
2 cents: Solution 2 seems more in line with schema.org and is more widely applicable, so I would favor that. We should rely on `propertyID` to identify the property we are talking about, e.g.: `"propertyID": "sc:bitrate"`

For properties that are not well described by sc:AudioObject, solution 2 can also be used, provided we find other vocabularies that define the right properties. `propertyID` can refer to the corresponding URLs.

(See examples 11 and 12 in the [schema.org/PropertyValue](schema.org/PropertyValue) documentation.)
